crates/types/src/lib.rs

```diff
@@ -1786,14 +1801,43 @@ impl From<wasmparser::MemoryType> for Memory {
 pub struct Tag {
     /// The event signature type.
     pub ty: TypeIndex,
+    /// The tag signature type.
+    pub signature: EngineOrModuleTypeIndex,
```

Comment:
Am I right in understanding that we need to store the `signature` (i.e., a canonicalized version of the type index) here in order to implement `TypeTrace` for `Tag`? And the `partial` function below is used for the case where we need to create such a `Tag` before knowing that canonicalized index?

Reply:
We should only ever need `signature`, but the problem is wasm-tools only exposes a bare module index. It is impossible to construct the `signature` in the two contexts where the `partial` constructor is currently used. I think the representation has to change in wasm-tools.

Reply:
Okay, I wasn't suggesting any changes to the definition of `Tag` here, only checking that my understanding for the need of these fields was correct.